src/gui/properties/peerlistwidget.cpp

```diff
@@ -405,7 +409,7 @@ void PeerListWidget::updatePeer(const BitTorrent::TorrentHandle *torrent, const
     m_listModel->setData(m_listModel->index(row, PeerListDelegate::DOWNLOADING_PIECE), downloadingFiles.join('\n'), Qt::ToolTipRole);
 
     if (m_resolver)
-        m_resolver->resolve(peerIp);
+        m_resolver->resolve(peerEndpoint.address.ip);
```

Comment:
Can you explain to me, we try to resolve peer host name each time we refresh the peer list? How often does this happen for the current torrent?

Reply:
sorry I cannot fully understand what you ask.

>How often does this happen for the current torrent?

Each time libtorrent posted an update.

Reply:
(Not related to this PR, just discussion and for those looking for contribute points)
I think (in current master branch) the reverse DNS resolution and country flag resolution are integrated badly in peerlistwidget and I suspect they are leading to high CPU usage when there are many peers (>=300) in list.
IMO when refreshing the peer list, we only need to resolve the new entries (peers) and leave the existing peers as is. The only moment we need to update the existing peers is when the user toggle the "Resolve peer host names" checkbox in Options from off to on, only then will we start to resolve the existing peers host name (and country flags). As for toggling from on to off, I would expect the resolved host name reverted to the original IP, not leaving them in host name format as we do it currently (and this is just my opinion).